### ISSUE TITLE
An $or-operator query with an $and-operator fails to construct properly.

### ISSUE THREAD

**Issue by jsprw** [mikro-orm/mikro-orm#247]
**Describe the bug**
If you query like:

```
const items = [{ prop1: "value1", prop2: "value2" }, { prop1: "value3", prop2: "value4" }]; 
const connection = await Database.getConnection();
return connection.em.getRepository(Entity).find({
       $or: items.map(i => ({
           $and: [{ prop1: i.prop1, prop2: i.prop2}]
      }))
});
```

... it will construct a query like:

```select `e0`.* from `entity` as `e0` where (((`e0`.`prop1` = 'value1' and `e0`.`prop2` = 'value2')) and ((`e0`.`prop1` = 'value3' and `e0`.`prop2` = 'value4')))```

... instead of the expected: 

```select `e0`.* from `entity` as `e0` where (((`e0`.`prop1` = 'value1' and `e0`.`prop2` = 'value2')) or ((`e0`.`prop1` = 'value3' and `e0`.`prop2` = 'value4')))```

**Stack trace**
No error, so no stack trace.

**To Reproduce**
See above.

**Expected behavior**
Create this query:

```select `e0`.* from `entity` as `e0` where (((`e0`.`prop1` = 'value1' and `e0`.`prop2` = 'value2')) or ((`e0`.`prop1` = 'value3' and `e0`.`prop2` = 'value4')))```

**Additional context**
It is related with the querybuilding using the incorrect synthax to append the group. I've created an additional repo where the query is constructed correctly, but I am not sure what effect the removal of that if has.

https://github.com/mikro-orm/mikro-orm/compare/dev...jsprw:patch-2

**Versions**

| Dependency | Version |
| - | - |
| node | v12.9.1 |
| typescript | v3.6.4 |
| mikro-orm | v3.0.0-rc |
| your-driver | mysql2 v2.0.1 |

**Comment by B4nan** [comment]
A bug indeed, this should be working as you expected, confirmed and replicated. 

But you are overcomplicating it a bit, you can pass the `items` array directly and it will work as you want:

```typescript
const items = [{ prop1: "value1", prop2: "value2" }, { prop1: "value3", prop2: "value4" }]; 
const connection = await Database.getConnection();
return connection.em.getRepository(Entity).find({
       $or: items
});
```

Multiple object keys are considered as ANDs, so `prop1: { $and: [1,2] }` is the same as `prop1: [1,2]`.

**Comment by jsprw** [comment]
> Multiple object keys are considered as ANDs, so prop1: { $and: [1,2] } is the same as prop1: [1,2].

Thank you! It will make the code easier to read!

**Comment by B4nan** [comment]
Fixed in `3.0.0-rc.1`
